src/picosvg/svg.py

```diff
@@ -266,22 +348,21 @@ def apply_style_attributes(self, inplace=False):
 
         return self
 
-    def xpath(self, xpath, el=None):
+    def xpath(self, xpath: str, el: etree.Element = None, expected_result_range=None):
         if el is None:
             el = self.svg_root
-        return el.xpath(xpath, namespaces={"svg": svgns()})
+        results = el.xpath(xpath, namespaces={"svg": svgns()})
+        if expected_result_range and len(results) not in expected_result_range:
+            raise ValueError(
+                f"Expected {xpath} matches in {expected_result_range}, {len(results)} results"
```

Comment:
I wonder, why is it a range and not just the maximum number of matching elements allowed (1 alwyas being the implicit minimum)?

Reply:
Because a range lets you express 0..N cleanly so you can easily say both exactly 1 or 0..1